src/lib/compress_hw.c

```diff
@@ -565,6 +589,7 @@ static int compress_hw_wait(void *data, int timeout_ms)
 
 struct compress_ops compress_hw_ops = {
 	.open = compress_hw_open,
```

Comment:
I don't think that we need two open functions. Keep only open_by_name version which covers overall functionality.

Reply:
I think @vinodkoul had a concern in removing existing API support. Do you mean that compress_open() should also jump to open_by_name() in compress_hw.c and name can be constructed using card/device in compress.c ?

Reply:
I think that @vinodkoul has only concerns about the exported API not the internal API which we define now.

Reply:
That is quite correct, exported APIs can't .... but no issue changing internal ones ;-)

Reply:
sure, I will update it.

Reply:
Done